circuit_knitting/cutting/instructions/move.py

```diff
@@ -0,0 +1,50 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Two-qubit instruction representing a swap + single-qubit reset."""
+from __future__ import annotations
+
+from qiskit.circuit import QuantumCircuit, Instruction
+
+
+class Move(Instruction):
+    """A two-qubit instruction representing a reset of the second qubit followed by a swap.
+
+    The desired effect of this instruction, typically, is to move the state of
+    the first qubit to the second qubit.  For this to work as expected, the
+    second incoming qubit must share no entanglement with the remainder of the
+    system.  If this qubit *is* entangled, then performing the reset operation
+    will in turn implement a quantum channel on the other qubit(s) with which
+    it is entangled, resulting in the partial collapse of those qubits.
+
+    Generally speaking, the most straightforward way to ensure that the second
+    qubit is not entangled is to prepare it in one of the following ways before
+    implementing the :class:`Move`:
+
+    - No operations since initialization.
```

Comment:
This is a great writeup, but I find these bullets, specifically, a bit confusing.

Maybe you can explicitly say in the first bullet "The qubit should contain no operations since circuit initialization" or something. I know it's more verbose, but it'll make it 100% clear we're talking about preparing this 2nd qubit.

I think the rest will be more clear if you add to the first point, but I am a bit confused about both `Measure` and `Reset` being the "preceding" operation. I would think only one could directly precede the `Move`. I would just try to make this a bit more clear.

I'm a bit confused on the last point too. This seems like it is an either/or with the other three items, but it's lumped in like all 4 are strong suggestions for preparing a `Move_2ndQ`.

Maybe drawing an example circuit in the docstring is appropriate here?

Reply:
I tried to improve this in 88f98d3cf6dd5d1e8a2a93a503d09fb9cbf68d61.  Rather than list all of the possibilities (which, actually, my previous list didn't even do exhaustively), I tried to focus on the two main ways in which you can satisfy the condition: use a fresh qubit as destination qubit OR use a qubit that was previously the source of a different move.  The two move instructions in the tutorial correspond to these cases, respectively.

Reply:
> Maybe drawing an example circuit or two in the docstring is appropriate here?

I'm going to look into how to draw circuits within docstrings, too.

Reply:
I pushed some further improvements in c716d9854788649b691db61cc7326341f8e9c6ef.